ShouldContainExactlyTest is flaky on Windows

I've noticed that the tests in ShouldContainExactlyTest sometimes fail on Windows machines.

The assertion failure is quite large and hard to see, but I think it's something to do with path separators `/` vs `\`. Which is weird, because that should be constant on a single machine. 

- https://scans.gradle.com/s/weonl3kvvhtqi ([job](https://github.com/kotest/kotest/actions/runs/10237567487/job/28321070983?pr=4256#step:8:873))

- https://scans.gradle.com/s/jp6bidtsxwxf4

```
> Task :kotest-assertions:kotest-assertions-core:jvmTest

com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest[jvm]

  containExactly should[jvm]

    Test include extras when too many[jvm] FAILED

    io.kotest.assertions.MultiAssertionError: The following 2 assertions failed:
    1) "Collection should contain exactly: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=23423, p=abc),
      but was: Blonde(a=woo, b=true, c=97821, p=abc),
      The following fields did not match:
        "a" expected: <"foo">, but was: <"woo">
            "c" expected: <23423>, but was: <97821>
    expected:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc)]>" should include substring "expected: Blonde(a=foo, b=true, c=23423, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$8.invokeSuspend(ShouldContainExactlyTest.kt:205)
    2) "Collection should contain exactly: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=23423, p=abc),
      but was: Blonde(a=woo, b=true, c=97821, p=abc),
      The following fields did not match:
        "a" expected: <"foo">, but was: <"woo">
            "c" expected: <23423>, but was: <97821>
    expected:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc)]>" should include substring "but was: Blonde(a=woo, b=true, c=97821, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$8.invokeSuspend(ShouldContainExactlyTest.kt:206)


    Test include missing and extras when not the right amount[jvm] FAILED

    io.kotest.assertions.MultiAssertionError: The following 2 assertions failed:
    1) "Collection should contain exactly: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96[915](https://github.com/kotest/kotest/actions/runs/10237567487/job/28321070983?pr=4256#step:8:916), p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=goo, b=true, c=51984, p=abc),
      but was: Blonde(a=woo, b=true, c=97821, p=abc),
      The following fields did not match:
        "a" expected: <"goo">, but was: <"woo">
            "c" expected: <51984>, but was: <97821>
    
    
     expected: Blonde(a=woo, b=true, c=97821, p=abc),
      but was: Blonde(a=goo, b=true, c=51984, p=abc),
      The following fields did not match:
        "a" expected: <"woo">, but was: <"goo">
            "c" expected: <97821>, but was: <51984>
    expected:<[Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]>" should include substring "expected: Blonde(a=goo, b=true, c=51984, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$10.invokeSuspend(ShouldContainExactlyTest.kt:253)
    2) "Collection should contain exactly: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=goo, b=true, c=51984, p=abc),
      but was: Blonde(a=woo, b=true, c=97821, p=abc),
      The following fields did not match:
        "a" expected: <"goo">, but was: <"woo">
            "c" expected: <51984>, but was: <97821>
    
    
     expected: Blonde(a=woo, b=true, c=97821, p=abc),
      but was: Blonde(a=goo, b=true, c=51984, p=abc),
      The following fields did not match:
        "a" expected: <"woo">, but was: <"goo">
            "c" expected: <97821>, but was: <51984>
    expected:<[Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]>" should include substring "but was: Blonde(a=woo, b=true, c=97821, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$10.invokeSuspend(ShouldContainExactlyTest.kt:254)


    Test exclude full print with warning on large collections[jvm] FAILED

    io.kotest.assertions.MultiAssertionError: The following 2 assertions failed:
    1) "Collection should contain exactly: [Blonde(a=foo, b=true, c=77, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)] but was: [Blonde(a=foo, b=true, c=1, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)]
    Some elements were missing: [Blonde(a=foo, b=true, c=77, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=1, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=2, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <2>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=3, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <3>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=4, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <4>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=5, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <5>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=6, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <6>, but was: <77>
    Printed first 5 similarities out of 20, (set the 'kotest.assertions.similarity.print.size' JVM property to see full output for similarity)(set the 'kotest.assertions.collection.enumerate.size' JVM property to see full output)" should include substring "expected: Blonde(a=foo, b=true, c=2, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$11.invokeSuspend(ShouldContainExactlyTest.kt:320)
    2) "Collection should contain exactly: [Blonde(a=foo, b=true, c=77, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)] but was: [Blonde(a=foo, b=true, c=1, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)]
    Some elements were missing: [Blonde(a=foo, b=true, c=77, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=1, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=2, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <2>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=3, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <3>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=4, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <4>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=5, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <5>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=6, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <6>, but was: <77>
    Printed first 5 similarities out of 20, (set the 'kotest.assertions.similarity.print.size' JVM property to see full output for similarity)(set the 'kotest.assertions.collection.enumerate.size' JVM property to see full output)" should include substring "but was: Blonde(a=foo, b=true, c=77, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$11.invokeSuspend(ShouldContainExactlyTest.kt:321)

[/[]

> Task :kotest-assertions:kotest-assertions-core:jvmMaxJdkTest
com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest[jvm]

  containExactly should[jvm]

    Test include extras when too many[jvm] FAILED

    io.kotest.assertions.MultiAssertionError: The following 2 assertions failed:
    1) "Collection should contain exactly: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=23423, p=abc),
      but was: Blonde(a=woo, b=true, c=97821, p=abc),
      The following fields did not match:
        "a" expected: <"foo">, but was: <"woo">
            "c" expected: <23423>, but was: <97821>
    expected:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc)]>" should include substring "expected: Blonde(a=foo, b=true, c=23423, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$8.invokeSuspend(ShouldContainExactlyTest.kt:205)
    2) "Collection should contain exactly: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=23423, p=abc),
      but was: Blonde(a=woo, b=true, c=97821, p=abc),
      The following fields did not match:
        "a" expected: <"foo">, but was: <"woo">
            "c" expected: <23423>, but was: <97821>
    expected:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=woo, b=true, c=97821, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc)]>" should include substring "but was: Blonde(a=woo, b=true, c=97821, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$8.invokeSuspend(ShouldContainExactlyTest.kt:206)


    Test include missing and extras when not the right amount[jvm] FAILED

    io.kotest.assertions.MultiAssertionError: The following 2 assertions failed:
    1) "Collection should contain exactly: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=[969](https://github.com/kotest/kotest/actions/runs/10237567487/job/28321070983?pr=4256#step:8:970)15, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=goo, b=true, c=51984, p=abc),
      but was: Blonde(a=woo, b=true, c=[978](https://github.com/kotest/kotest/actions/runs/10237567487/job/28321070983?pr=4256#step:8:979)21, p=abc),
      The following fields did not match:
        "a" expected: <"goo">, but was: <"woo">
            "c" expected: <51984>, but was: <97821>
    
    
     expected: Blonde(a=woo, b=true, c=97821, p=abc),
      but was: Blonde(a=goo, b=true, c=51984, p=abc),
      The following fields did not match:
        "a" expected: <"woo">, but was: <"goo">
            "c" expected: <97821>, but was: <51984>
    expected:<[Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]>" should include substring "expected: Blonde(a=goo, b=true, c=51[984](https://github.com/kotest/kotest/actions/runs/10237567487/job/28321070983?pr=4256#step:8:985), p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$10.invokeSuspend(ShouldContainExactlyTest.kt:253)
    2) "Collection should contain exactly: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] but was: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]
    Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=goo, b=true, c=51984, p=abc),
      but was: Blonde(a=woo, b=true, c=97821, p=abc),
      The following fields did not match:
        "a" expected: <"goo">, but was: <"woo">
            "c" expected: <51984>, but was: <97821>
    
    
     expected: Blonde(a=woo, b=true, c=97821, p=abc),
      but was: Blonde(a=goo, b=true, c=51984, p=abc),
      The following fields did not match:
        "a" expected: <"woo">, but was: <"goo">
            "c" expected: <97821>, but was: <51984>
    expected:<[Blonde(a=woo, b=true, c=97821, p=abc), Blonde(a=goo, b=true, c=51984, p=abc)]> but was:<[Blonde(a=foo, b=true, c=23423, p=abc), Blonde(a=hoo, b=true, c=96915, p=abc)]>" should include substring "but was: Blonde(a=woo, b=true, c=97821, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$10.invokeSuspend(ShouldContainExactlyTest.kt:254)


    Test exclude full print with warning on large collections[jvm] FAILED

    io.kotest.assertions.MultiAssertionError: The following 2 assertions failed:
    1) "Collection should contain exactly: [Blonde(a=foo, b=true, c=77, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)] but was: [Blonde(a=foo, b=true, c=1, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)]
    Some elements were missing: [Blonde(a=foo, b=true, c=77, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=1, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=2, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <2>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=3, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <3>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=4, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <4>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=5, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <5>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=6, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <6>, but was: <77>
    Printed first 5 similarities out of 20, (set the 'kotest.assertions.similarity.print.size' JVM property to see full output for similarity)(set the 'kotest.assertions.collection.enumerate.size' JVM property to see full output)" should include substring "expected: Blonde(a=foo, b=true, c=2, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$11.invokeSuspend(ShouldContainExactlyTest.kt:320)
    2) "Collection should contain exactly: [Blonde(a=foo, b=true, c=77, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)] but was: [Blonde(a=foo, b=true, c=1, p=abc), Blonde(a=foo, b=true, c=2, p=abc), Blonde(a=foo, b=true, c=3, p=abc), Blonde(a=foo, b=true, c=4, p=abc), Blonde(a=foo, b=true, c=5, p=abc), Blonde(a=foo, b=true, c=6, p=abc), Blonde(a=foo, b=true, c=7, p=abc), Blonde(a=foo, b=true, c=8, p=abc), Blonde(a=foo, b=true, c=9, p=abc), Blonde(a=foo, b=true, c=10, p=abc), Blonde(a=foo, b=true, c=11, p=abc), Blonde(a=foo, b=true, c=12, p=abc), Blonde(a=foo, b=true, c=13, p=abc), Blonde(a=foo, b=true, c=14, p=abc), Blonde(a=foo, b=true, c=15, p=abc), Blonde(a=foo, b=true, c=16, p=abc), Blonde(a=foo, b=true, c=17, p=abc), Blonde(a=foo, b=true, c=18, p=abc), Blonde(a=foo, b=true, c=19, p=abc), Blonde(a=foo, b=true, c=20, p=abc), ...and 1 more (set the 'kotest.assertions.collection.print.size' JVM property to see more / less items)]
    Some elements were missing: [Blonde(a=foo, b=true, c=77, p=abc)] and some elements were unexpected: [Blonde(a=foo, b=true, c=1, p=abc)]
    
    Possible matches:
    
     expected: Blonde(a=foo, b=true, c=2, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <2>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=3, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <3>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=4, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <4>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=5, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <5>, but was: <77>
    
     expected: Blonde(a=foo, b=true, c=6, p=abc),
      but was: Blonde(a=foo, b=true, c=77, p=abc),
      The following fields did not match:
        "c" expected: <6>, but was: <77>
    Printed first 5 similarities out of 20, (set the 'kotest.assertions.similarity.print.size' JVM property to see full output for similarity)(set the 'kotest.assertions.collection.enumerate.size' JVM property to see full output)" should include substring "but was: Blonde(a=foo, b=true, c=77, p=a/b/c),"
       at com.sksamuel.kotest.matchers.collections.ShouldContainExactlyTest$1$11.invokeSuspend(ShouldContainExactlyTest.kt:321)

[/[]
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended



> Task :kotest-assertions:kotest-assertions-core:jvmMaxJdkTest
2078 tests completed, 3 failed

FAILURE: Executed 2078 tests in 42.8s (3 failed)
```


Related to #4113

--- a/kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldContainExactlyTest.kt
+++ b/kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldContainExactlyTest.kt
@@ -202,8 +202,8 @@ class ShouldContainExactlyTest : WordSpec() {
                )
 
             message.shouldContain("Possible matches:")
-            message.shouldContain("expected: Blonde(a=foo, b=true, c=23423, p=a/b/c),")
-            message.shouldContain("but was: Blonde(a=woo, b=true, c=97821, p=a/b/c),")
+            message.shouldContain("expected: Blonde(a=foo, b=true, c=23423, p=$expectedPath),")
+            message.shouldContain("but was: Blonde(a=woo, b=true, c=97821, p=$expectedPath),")
             message.shouldContain("The following fields did not match:")
             message.shouldContain("\"a\" expected: <\"foo\">, but was: <\"woo\">")
             message.shouldContain("\"c\" expected: <23423>, but was: <97821>")
@@ -249,8 +249,8 @@ class ShouldContainExactlyTest : WordSpec() {
                   |Some elements were missing: [Blonde(a=woo, b=true, c=97821, p=$expectedPath), Blonde(a=goo, b=true, c=51984, p=$expectedPath)] and some elements were unexpected: [Blonde(a=foo, b=true, c=23423, p=$expectedPath), Blonde(a=hoo, b=true, c=96915, p=$expectedPath)]
                """.trimMargin()
             message.shouldContain("Possible matches:")
-            message.shouldContain("expected: Blonde(a=goo, b=true, c=51984, p=a/b/c),")
-            message.shouldContain("but was: Blonde(a=woo, b=true, c=97821, p=a/b/c),")
+            message.shouldContain("expected: Blonde(a=goo, b=true, c=51984, p=$expectedPath),")
+            message.shouldContain("but was: Blonde(a=woo, b=true, c=97821, p=$expectedPath),")
             message.shouldContain("The following fields did not match:")
             message.shouldContain("\"a\" expected: <\"goo\">, but was: <\"woo\">")
             message.shouldContain("\"c\" expected: <51984>, but was: <97821>")
@@ -314,8 +314,8 @@ class ShouldContainExactlyTest : WordSpec() {
                   |Some elements were missing: [Blonde(a=foo, b=true, c=77, p=$expectedPath)] and some elements were unexpected: [Blonde(a=foo, b=true, c=1, p=$expectedPath)]
                """.trimMargin()
                message.shouldContain("Possible matches:")
-               message.shouldContain("expected: Blonde(a=foo, b=true, c=2, p=a/b/c),")
-               message.shouldContain("but was: Blonde(a=foo, b=true, c=77, p=a/b/c),")
+               message.shouldContain("expected: Blonde(a=foo, b=true, c=2, p=$expectedPath),")
+               message.shouldContain("but was: Blonde(a=foo, b=true, c=77, p=$expectedPath),")
                message.shouldContain("The following fields did not match:")
                message.shouldContain("\"c\" expected: <2>, but was: <77>")
             message shouldContain "Printed first 5 similarities out of 20, (set the 'kotest.assertions.similarity.print.size' JVM property to see full output for similarity)"

